Metric_Publisher/Functions/Metrics.py

```diff
@@ -0,0 +1,56 @@
+from Metric_Publisher.Functions.Metric import Metric
+from contextlib import contextmanager
+import time
+
+
+class TimingMetric(Metric):
+    """Timing metric that calculates the time the metric has run."""
+
+    def __init__(self, metric_name: str, unit: str):
+        super().__init__(metric_name, unit)
+
+    @contextmanager
+    def timeit_metric(self):
+        """Context manager used to calculate the time the method has run."""
+        start_time = 0
+        try:
+            start_time = time.time()
+            yield
+        finally:
+            self._value = time.time() - start_time
```

Comment:
I think just timeit() works for a method name. Also instead of a separate context manager inside this class I would prefer one of two things:

1) the TimingMetric class implements the context manager methods -__enter__ and __exit__ and we can use the metric object directly as the context manager instead of a separate function

2) make this a utility function somewhere instead of in the class, and have it create the metric obj